pkg/option/config.go

```diff
@@ -2615,7 +2629,7 @@ func (c *DaemonConfig) Validate(vp *viper.Viper) error {
 			int64(defaults.KVstoreLeaseMaxTTL.Seconds()))
 	}
 
-	if c.WriteCNIConfigurationWhenReady != "" && c.ReadCNIConfiguration == "" {
+	if c.WriteCNIConfigurationWhenReady == "" && c.ReadCNIConfiguration != "" {
 		return fmt.Errorf("%s must be set when using %s", ReadCNIConfiguration, WriteCNIConfigurationWhenReady)
```

Comment:
Fair enough -- we don't really need the error check. It would just catch an unlikely misconfiguration, essentially equivalent to `cp <src>` without the matching `<dst>`